lib/acl.c

```diff
@@ -45,12 +45,31 @@ enum {
 
 /* Callback function for when there's no classification match. */
 static inline int
-drop_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
-	__attribute__((unused)) struct gatekeeper_if *iface)
+match_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct gatekeeper_if *iface)
 {
 	unsigned int i;
-	for (i = 0; i < num_pkts; i++)
-		rte_pktmbuf_free(pkts[i]);
+	for (i = 0; i < num_pkts; i++) {
+		unsigned int j;
+		bool free_pkt = false;
+		/* Start from the first one. */
+		for (j = 1; j < iface->acl_func_count; j++) {
+			int ret = iface->ext_funcs[j](pkts[i], iface);
+			if (ret == -ENOENT)
+				free_pkt = true;
+			else {
+				/*
+				 * The packet is either freed when error
+				 * happens or successfully matched.
+				 */
+				free_pkt = false;
+				break;
+			}
+		}
+
+		if (free_pkt)
+			rte_pktmbuf_free(pkts[i]);
```

Comment:
Before releasing packets here, we need to log an informative message describing the packet that is about to be dropped. This message must be a warning.

The reason for the log is that a packet has reached a Gatekeeper server, and Gatekeeper doesn't know what to do with this packet. If attackers are able to send these packets, they may be able to slow Gatekeeper down since Gatekeeper does a lot of processing to eventually discard these packets. A comment in the code should explain this too.